providers/amazon/src/airflow/providers/amazon/aws/log/cloudwatch_task_handler.py

```diff
@@ -275,9 +299,28 @@ def close(self):
         # Mark closed so we don't double write if close is called twice
         self.closed = True
 
-    def _read_remote_logs(
-        self, task_instance, try_number, metadata=None
-    ) -> tuple[LogSourceInfo, LogMessages]:
+    def _read_remote_logs(self, task_instance, try_number, metadata=None) -> LegacyLogResponse | LogResponse:
         stream_name = self._render_filename(task_instance, try_number)
-        messages, logs = self.io.read(stream_name, task_instance)
-        return messages, logs or []
+        messages = [
+            f"Reading remote log from Cloudwatch log_group: {self.io.log_group} log_stream: {stream_name}"
+        ]
+
+        logs: list[str] | list[Generator[str, None, None]]
+        try:
+            events = self.io.get_cloudwatch_logs(stream_name, task_instance)
+            if SUPPORT_STREAM_BASED_READ:
```

Comment:
Read on CloudwatchTaskHandler should only ever be called for old Airflow Core versions (Newer versions use `CloudwatchRemoteIO` directly instead -- do we need this here?